src/Microsoft.AspNet.Mvc.Core/ParameterBinding/ModelBindingHelper.cs

```diff
@@ -381,5 +387,54 @@ public static string CreatePropertyModelName(string prefix, string propertyName)
                 return prefix + "." + propertyName;
             }
         }
+
+        internal static void ClearModelStateDictionaryForModel(
```

Comment:
I am with Ryan on this one. The only _minor_ problem I see is that the class does not seem to be the right place of exposing a method related to model state dictionary. I am ok making this public.